pkg/controller/podgc/gc_controller.go

```diff
@@ -96,11 +96,7 @@ func (gcc *PodGCController) gc() {
 	terminatedPodCount := len(terminatedPods)
 	sort.Sort(byCreationTimestamp(terminatedPods))
 
-	deleteCount := terminatedPodCount - gcc.threshold
-
-	if deleteCount > terminatedPodCount {
-		deleteCount = terminatedPodCount
-	}
+	deleteCount := terminatedPodCount
```

Comment:
I believe the intent here is to delete the N oldest pods that are above the gc threshold. So if you have 1 terminated pod and the threshold is 5, don't delete it. If you have 6, sort the terminated pods by creation timestamp and delete the first one.